roachtest: add acceptance/decommission

petermattis opened the pull request:
Move the decommission acceptance test to a new acceptance/decommission
roachtest.

Fixes #29151

Release note: None

petermattis committed:
roachtest: add acceptance/decommission

Move the decommission acceptance test to a new acceptance/decommission
roachtest.

Fixes #29151

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/29488)
<!-- Reviewable:end -->

petermattis commented:
The first commit is https://github.com/cockroachdb/cockroach/pull/29470.

This removes the last usage of `RunLocal` from `pkg/acceptance` tests.

petermattis commented:
`github-pull-request-make` is return an empty pkg from `pkgsFromDiff` which is causing `test` and `testrace` CI runs to fail. Taking a look.

petermattis committed:
github-pull-request-make: don't add packages with empty names

When deleting a go test it was possible to add a package with an empty
name. This in turn would cause CI failure as the top-level cockroach
package contains no Go sources.

Release note: None

petermattis commented:
Added a commit which should fix the `github-pull-request-make` problem. We apparently don't delete tests very often.

tbg reviewed (commented):
> Reviewed 4 of 4 files at r1, 1 of 1 files at r2, 5 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29488)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/decommission.go, line 261 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzUoV3cb9fxNMCfjE:-LLZzUoV3cb9fxNMCfjF:bwjbaf2) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L261)):*
> ```Go
> 		extraArgs ...string,
> 	) (string, error) {
> 		args := []string{"ssh", c.makeNodes(c.Node(runNode)), "--", cockroach}
> ```

Isn't this quite roundabout? Why not `c.RunE(ctx, c.Node(runNode), "./cockroach", ...)`


I think this might also reintroduce these hangs about passing stdout into child processes.
___
*[pkg/cmd/roachtest/decommission.go, line 273 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzr5i5Q5TIhepdEwM:-LLZzr5i5Q5TIhepdEwN:b-ynpvxi) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L273)):*
> ```Go
> 		return buf.String(), err
> 	}
> ```

Any changes below this line that I should review? I skimmed it and it looks like the old test.


<!-- Sent from Reviewable.io -->

petermattis reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29488)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/decommission.go, line 261 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzUoV3cb9fxNMCfjE:-LL_33yZDjdyh-h1pe5n:b-jw6fwy) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L261)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

Isn't this quite roundabout? Why not `c.RunE(ctx, c.Node(runNode), "./cockroach", ...)`


I think this might also reintroduce these hangs about passing stdout into child processes.
</blockquote></details>

Hmm, good point. The reason I'm not using `RunE` is that I need to capture stdout.  I guess I could avoid the teeing to `c.l.stdout`, though that is really nice for debugging. Let me take another look at this.
___
*[pkg/cmd/roachtest/decommission.go, line 273 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzr5i5Q5TIhepdEwM:-LL_3A5X9E2ZI2neUnJs:bnw7l6q) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L273)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

Any changes below this line that I should review? I skimmed it and it looks like the old test.
</blockquote></details>

No, everything below here was directly adapted from the old test.


<!-- Sent from Reviewable.io -->

tschottdorf commented:
We have `RunWithBuffer`, doesn't that help?

On Tue, Sep 4, 2018 at 4:52 PM Peter Mattis <notifications@github.com>
wrote:

> *@petermattis* commented on this pull request.
>
> *Reviewable <https://reviewable.io/reviews/cockroachdb/cockroach/29488>*
> status: [image: :shipit:] complete! 0 of 0 LGTMs obtained
>
> ------------------------------
>
> *pkg/cmd/roachtest/decommission.go, line 261 at r3
> <https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzUoV3cb9fxNMCfjE:-LL_33yZDjdyh-h1pe5n:b-jw6fwy>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L261>):*
> *Previously, tschottdorf (Tobias Schottdorf) wrote…*
>
> Isn't this quite roundabout? Why not c.RunE(ctx, c.Node(runNode),
> "./cockroach", ...)
>
> I think this might also reintroduce these hangs about passing stdout into
> child processes.
>
> Hmm, good point. The reason I'm not using RunE is that I need to capture
> stdout. I guess I could avoid the teeing to c.l.stdout, though that is
> really nice for debugging. Let me take another look at this.
> ------------------------------
>
> *pkg/cmd/roachtest/decommission.go, line 273 at r3
> <https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzr5i5Q5TIhepdEwM:-LL_3A5X9E2ZI2neUnJs:bnw7l6q>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L273>):*
> *Previously, tschottdorf (Tobias Schottdorf) wrote…*
>
> Any changes below this line that I should review? I skimmed it and it
> looks like the old test.
>
> No, everything below here was directly adapted from the old test.
>
> —
> You are receiving this because your review was requested.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/29488#pullrequestreview-152105582>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE135IuBt-D7Y9PMKXAu9pgC46dozyo2ks5uXpOqgaJpZM4WX32f>
> .
>
-- 

-- Tobias

petermattis reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29488)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/decommission.go, line 261 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LLZzUoV3cb9fxNMCfjE:-LL_5L9Q30_fyyJvrBU2:b-2ihxl7) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ecf6678fb8b545f6644c1ec830fdd8f382e4c59/pkg/cmd/roachtest/decommission.go#L261)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Hmm, good point. The reason I'm not using `RunE` is that I need to capture stdout.  I guess I could avoid the teeing to `c.l.stdout`, though that is really nice for debugging. Let me take another look at this.
</blockquote></details>

Ok, I've changed this to use `cluster.RunWithBuffer`. No more fancy teeing. Oh well. 


<!-- Sent from Reviewable.io -->

benesch reviewed (approved):
> Reviewed 1 of 1 files at r2, 5 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29488)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/github-pull-request-make/main.go, line 119 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LL_UMJMARbTpaNcDKii:-LL_UMJMARbTpaNcDKij:bckn27m) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8c050fc4eef64b3243671a6456e02792f57cf9c/pkg/cmd/github-pull-request-make/main.go#L119)):*
> ```Go
> 				case len(curTestName) > 0:
> 					if !(curPkgName == "build" && curTestName == "TestStyle") {
> 						fmt.Printf("adding test: name=%s pkg=%s\n", curTestName, curPkgName)
> ```

Did you mean to leave this in?


<!-- Sent from Reviewable.io -->

petermattis reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29488)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/github-pull-request-make/main.go, line 119 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/29488#-LL_UMJMARbTpaNcDKii:-LL_YU9l8ESlFpGyMWjk:bu2j3sr) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8c050fc4eef64b3243671a6456e02792f57cf9c/pkg/cmd/github-pull-request-make/main.go#L119)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Did you mean to leave this in?
</blockquote></details>

Nope. Removed.


<!-- Sent from Reviewable.io -->

petermattis commented:
bors r=benesch,tschottdorf

petermattis commented:
(bors crash, I think)

bors r=benesch,tschottdorf

petermattis commented:
bors r=benesch,tschottdorf
